src/NuGet.Core/NuGet.Protocol/LegacyFeed/ODataServiceDocumentResourceV2Provider.cs

```diff
@@ -12,11 +12,12 @@
 
 namespace NuGet.Protocol
 {
-    public class ODataServiceDocumentResourceV2Provider : ResourceProvider
+    public class ODataServiceDocumentResourceV2Provider : ResourceProvider, IDisposable
```

Comment:
I agree with your comment, Andy. I learnt that there are so many classes that inherit `ResourceProvider` class which don't implement `IDisposable` interface implicitly/explicitly. Changing one derived class of `ResourceProvider` to implement `IDisposable` doesn't add much value because the caller cannot call `Dispose` method without an explicit cast to `IDisposable`. I have reverted the changes from `NuGet.Protocol` project and added `justification` to the suppressions as they require a bit more work to fix CA1001 warnings.